x-pack/plugins/security_solution/server/endpoint/ingest_integration.ts

```diff
@@ -8,9 +8,63 @@ import { Logger } from '../../../../../src/core/server';
 import { NewPackageConfig } from '../../../ingest_manager/common/types/models';
 import { factory as policyConfigFactory } from '../../common/endpoint/models/policy_config';
 import { NewPolicyData } from '../../common/endpoint/types';
-import { ManifestManager, ManifestSnapshot } from './services/artifacts';
+import { ManifestManager } from './services/artifacts';
+import { Manifest } from './lib/artifacts';
 import { reportErrors, ManifestConstants } from './lib/artifacts/common';
-import { ManifestSchemaVersion } from '../../common/endpoint/schema/common';
+import { InternalArtifactSchema } from './schemas/artifacts';
+import { manifestDispatchSchema } from '../../common/endpoint/schema/manifest';
+
+const getManifest = async (logger: Logger, manifestManager: ManifestManager): Promise<Manifest> => {
+  let manifest: Manifest | null = null;
+
+  try {
+    manifest = await manifestManager.getLastComputedManifest(ManifestConstants.SCHEMA_VERSION);
+
+    // If we have not yet computed a manifest, then we have to do so now. This should only happen
+    // once.
+    if (manifest == null) {
+      // New computed manifest based on current state of exception list
+      const newManifest = await manifestManager.buildNewManifest(ManifestConstants.SCHEMA_VERSION);
+      const diffs = newManifest.diff(Manifest.getDefault(ManifestConstants.SCHEMA_VERSION));
```

Comment:
Would this always be empty since this is the first manifest? Or are diffs created since they are going from nothing to something?

Reply:
@alexk307 There will be a diff that just contains all 3 of the artifacts.